lib/metrics.js

```diff
@@ -0,0 +1,107 @@
+then your use, modification, or distribution of it requires the prior
+written permission of Adobe.
+*/
+
+'use strict';
+
+const { readMetric, formatMetrics } = require('./utils');
+
+/**
+ * Reads metrics from `/sys/fs/cgroup/memory`
+ * @returns Two asyncronous functions `containerUsage` and `containerUsagePercentage`
+ */
+function memory() {
+    return {
+        containerUsage: async () => {
+            const rss = await readMetric('memory/memory.stat');
+            const kmemUsage = await readMetric('memory/memory.kmem.usage_in_bytes');
+            if (typeof(rss) !== "number" || typeof(kmemUsage) !== "number" || (isNaN(rss)) || (isNaN(kmemUsage)) ) {
+                throw Error(`One or more metrics are malformed. rss: ${rss}, kmemUsage: ${kmemUsage}`);
+            }
+            return (rss + kmemUsage);
+        },
+        containerUsagePercentage: async (containerUsage=false) => {
+            if (!(containerUsage)) {
+                const rss = await readMetric('memory/memory.stat');
+                const kmemUsage = await readMetric('memory/memory.kmem.usage_in_bytes');
+                if (typeof(rss) !== "number" || typeof(kmemUsage) !== "number" || (isNaN(rss)) || (isNaN(kmemUsage)) ) {
+                    throw Error(`One or more metrics are malformed. rss: ${rss}, kmemUsage: ${kmemUsage}`);
+                }
+                containerUsage = (rss + kmemUsage);
+            }
+            const limit = await readMetric('memory/memory.limit_in_bytes');
+            if (typeof(containerUsage) !== "number" || typeof(limit) !== "number" || (isNaN(containerUsage)) || (isNaN(limit)) ) {
+                throw Error(`One or more metrics are malformed. containerUsage: ${containerUsage}, limit: ${limit}`);
+            }
+            return ((containerUsage) / limit);
+        }
+    }
+}
+
+/**
+ * Reads metrics from `/sys/fs/cgroup/cpuacct`
+ * @returns Three asyncronous functions `usage` and `stat`
+ */
+function cpu() {
+    return {
+
+        // returns a an object { timeSinceContainerNS: <cpuacct/cpuacct.usage>, timestamp: <timestamp>}
+        usage: async () => {
+            const timeSinceContainerNS = await readMetric('cpuacct/cpuacct.usage');
+            return {
+                timeSinceContainerNS: timeSinceContainerNS,
+                timestamp: Date.now()
+            }
+        },
+
+        // returns an object {user: <user-cpu-amount> , system: <system-cpu-amount>, timestamp: <timestamp>}
+        stat: async () => {
+            const stat = await readMetric('cpuacct/cpuacct.stat');
+            return Object.assign( stat, {
+                timestamp: Date.now()
+            })
+        },
+
+        calculatedUsage: (usageFirst, usageSecond) => {
+            const totalTime = usageSecond.timestamp - usageFirst.timestamp;
+            const deltaNS = usageSecond.timeSinceContainerNS - usageFirst.timeSinceContainerNS;
+            return deltaNS / totalTime * 100
```

Comment:
The timestamps are in milliseconds, while the measurement is in nano seconds. I think it has to be adjusted.

Would be great to test this actually works with a little sample app and e.g. compare with `docker stats`.

Reply:
Sounds like a good idea. Should I use a library to mock a docker container?

Reply:
I was thinking of manual testing first. Because this isn't well documented...

Then your existing unit tests would be fine I guess, just the numbers would need adjustment.